### PR TITLE
Unify all our Amlogic devices

### PR DESCRIPTION
**Pull request by Raybuntu**
With this PR we can use only one S905.tar for S905, Odroid_C2, LePotato and KVIM and one S912.tar for S912 and KVIM2.
We can easily scale up now since we don't have to waste buildtime for 6 seperate clean builds but 2.

Even though only one S905.tar is created we still create 4 img.gz so that the user can flash for instance a LePotato image without having to do an additional/complicated u-boot flash.

This idea could also be extended to build several S905.img.gz with different multi-dtb's to give the user a proper out-of-the-box experience. We can also easily add and support more builds for SBC's with custom u-boot without any extra buildtime effort.

This will require a lot of testing.

### CODE REVIEW

**Comment by Raybuntu** [comment]
https://mega.nz/#F!Zw4zUAZY!x7z5YkLCY5xGlp7i7rYrlg

Here are the builds. I've tested fresh installs of LePotato and Odroid C2 and also tar updates from 8.95.5 with both LePotato and C2. All is working as expected!

File: projects/Amlogic/bootloader/mkbootimg [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2018-present Team CoreELEC (https://coreelec.org)
+
+DTB_PATH="arch/$TARGET_KERNEL_ARCH/boot/dts/amlogic"
```
Comment (vpeter4) [inline projects/Amlogic/bootloader/mkbootimg]: Set path only for DTBIMG_PATH and reuse it in DTB_PATH?

Comment (Raybuntu) [reply projects/Amlogic/bootloader/mkbootimg]: Thanks you are right

File: projects/Amlogic/packages/u-boot/package.mk [hunk 2]
```diff
@@ -1,86 +1,53 @@
...
-      "Odroid_C2")
-        cp -av $PKG_BUILD/u-boot.bin $INSTALL/usr/share/bootloader/u-boot
-        find_file_path splash/boot-logo.bmp.gz && cp -av ${FOUND_PATH} $INSTALL/usr/share/bootloader
-        ;;
-      "KVIM"*|"LePotato")
-        cp -av $PKG_BUILD/fip/u-boot.bin.sd.bin $INSTALL/usr/share/bootloader/u-boot
-        ;;
-    esac
+    for PKG_SUBDEVICE in $SUBDEVICES; do
+        find_file_path bootloader/${PKG_SUBDEVICE}_boot.ini && cp -av ${FOUND_PATH} $INSTALL/usr/share/bootloader
```
Comment (vpeter4) [inline projects/Amlogic/packages/u-boot/package.mk]: Two spaces extra for indent?

File: scripts/image [hunk 3]
```diff
@@ -335,32 +335,65 @@ if [ "$1" = "release" -o "$1" = "mkimage" -o "$1" = "amlpkg" -o "$1" = "noobs" ]
...
+          RELEASE_DIR=$RELEASE_DIR \
+          UUID_STORAGE="$(uuidgen)" \
+          DISTRO_BOOTLABEL="$DISTRO_BOOTLABEL" \
+          DISTRO_DISKLABEL="$DISTRO_DISKLABEL" \
+          UBOOT_SYSTEM="$UBOOT_SYSTEM" \
+          UBOOT_VERSION="$UBOOT_VERSION" \
+          EXTRA_CMDLINE="$EXTRA_CMDLINE" \
+          SYSTEM_SIZE="$SYSTEM_SIZE" \
+          SYSTEM_PART_START="$SYSTEM_PART_START" \
+	  SUBDEVICE=$SUBDEVICE \
```
Comment (vpeter4) [inline scripts/image]: indent

File: packages/linux/package.mk [hunk 4]
```diff
@@ -220,40 +220,16 @@ make_target() {
...
-        cp arch/$TARGET_KERNEL_ARCH/boot/dtb.img $INSTALL/usr/share/bootloader/dtb.img 2>/dev/null || :
-      fi
-    else
-      for dtb in arch/$TARGET_KERNEL_ARCH/boot/dts/*.dtb arch/$TARGET_KERNEL_ARCH/boot/dts/*/*.dtb; do
-        if [ -f $dtb ]; then
-          cp -v $dtb $INSTALL/usr/share/bootloader
-        fi
-      done
+    if [ -d arch/$TARGET_KERNEL_ARCH/boot/dts/amlogic ]; then
+        cp arch/$TARGET_KERNEL_ARCH/boot/*dtb.img $INSTALL/usr/share/bootloader/ 2>/dev/null || :
```
Comment (arthur-liberman) [inline packages/linux/package.mk]: indent here.

File: scripts/mkimage [hunk 5]
```diff
@@ -30,6 +30,9 @@ DISK_START_PADDING=$(( (${SYSTEM_PART_START} + 2048 - 1) / 2048 ))
 DISK_GPT_PADDING=1
 DISK_SIZE=$(( ${DISK_START_PADDING} + ${SYSTEM_SIZE} + ${STORAGE_SIZE} + ${DISK_GPT_PADDING} ))
 DISK_BASENAME="${TARGET_IMG}/${IMAGE_NAME}"
+if [ -n "$SUBDEVICE" ]; then
+  DISK_BASENAME="$TARGET_IMG/$IMAGE_NAME-$SUBDEVICE"
```
Comment (vpeter4) [inline scripts/mkimage]: DISK_BASENAME="$DISK_BASENAME-$SUBDEVICE" ?

File: projects/Amlogic/bootloader/update.sh [hunk 6]
```diff
@@ -128,4 +143,15 @@ if [ -f $BOOT_ROOT/aml_autoscript ]; then
...
+    echo "Updating Legacy Kernel name..."
+    cp -p $BOOT_ROOT/@LEGACY_KERNEL_NAME@ $BOOT_ROOT/@KERNEL_NAME@
+    cp -p $BOOT_ROOT/@LEGACY_KERNEL_NAME@.md5 $BOOT_ROOT/@KERNEL_NAME@.md5
+fi
+
+if [ ! -f $BOOT_ROOT/dtb.img -a -f $BOOT_ROOT/@LEGACY_DTB_NAME@ ]; then
+    echo "Updating Legacy dtb name..."
+    cp -p $BOOT_ROOT/@LEGACY_DTB_NAME@ $BOOT_ROOT/dtb.img
+fi
+
```
Comment (arthur-liberman) [inline projects/Amlogic/bootloader/update.sh]: Indents here

**Pull request merged**